Fixes extra newlines are added after switch case statements on strings

agandavarapu-rythmos opened the pull request:
This will remove the extra line before return after case_colon

If in switch "case" is having only one line and that too "return". This change will make sure that new line will not be added before return event though following option is set
nl_before_return = true

puppalasatish committed:
Fixes issue #1257

This will remove the extra line before return after case_colon

If in switch "case" is having only one line and that too "return" this change will make sure that new line will not be added before return event though following option is set
nl_before_return = true

hschlichter commented on tests/output/staging/60008-UNI-17253.cs:
```diff
@@ -1,4 +1,4 @@
-// It adds an extra newline after each case 
+// It adds an extra newline after each case
```
This change should be reverted. Creates noise.

spuppala-rythmos replied:
if we don't remove space in the comment , this test case will always fail.
so changed the output file.

NOTE :: uncrustify will remove the extra spaces in the end of a comment line

hschlichter commented on src/newlines.cpp:
```diff
@@ -1962,7 +1962,7 @@ static void newline_before_return(chunk_t *start)
    chunk_t *pc = chunk_get_prev(nl);
    if (  !pc
-      || (pc->type == CT_BRACE_OPEN || pc->type == CT_VBRACE_OPEN))
+      || (pc->type == CT_BRACE_OPEN || pc->type == CT_VBRACE_OPEN || pc->type == CT_CASE_COLON))
```
This only fixes the problem of `return` directly after a `:` So it doesn't really fix the problem.
Only the case of
```
    case "a":
        return foo;
```
will be fixed.

But if we add this to the test case(which we should) it still fails
```
    case "c":
        printf("hello world\n");
        return 0;
```
resulting in 
```
    case "c":
        printf("hello world\n");

        return 0;
```

puppalasatish committed:
Fixes issue # 1257

Incorporated comments , which fixes othe scenario

puppalasatish committed:
fixes issue # 1257

Added few extra test scenarios
staging updated for other stories
removed 10054 as part of story #1236
removed 60036 as part of story #1292

spuppala-rythmos committed:
Merge branch 'master' into fixes/Extra-newlines-are-added-after-switch-case-statements-on-strings
